### PR TITLE
[CALCITE-1764] Adding ordering field to sort spec

### PR DESCRIPTION
**Pull request by b-slim**

### CODE REVIEW

**Comment by b-slim** [comment]
@jcamachor can you please check this.

File: druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java [hunk 1]
```diff
@@ -634,9 +634,19 @@ protected QuerySpec getQuery(RelDataType rowType, RexNode filter, List<RexNode>
         ImmutableList.Builder<JsonCollation> colBuilder =
             ImmutableList.builder();
         for (Pair<Integer, Direction> p : Pair.zip(collationIndexes, collationDirections)) {
+          boolean isNumericSort = false;
```
Comment (jcamachor) [inline druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java]: Why don't we extract the _types_ in the loop in L464 into a list  _collationTypes_, we pass the list to _getQuery_ method, and then just use the following check?

``if (collationTypes.get(x).getType().getFamily() == SqlTypeFamily.NUMERIC) { ... } ``

That would be less ad-hoc than just using group by column position and then falling back to checking projects (plus it should work even if dimensions are not of character type in future).

Comment (b-slim) [inline druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java]: i thought about that, but looked a bigger more intrusive change than this one, but i can do it that way.

**Comment by b-slim** [comment]
@jcamachor checkout https://github.com/apache/calcite/pull/436/commits/b147b3f57bcaa96d131c9684ba2b6ddba36e63ee

**Comment by b-slim** [comment]
@jcamachor all the tests passed for this one

**Pull request closed without merging**